wordpress-mobile/WordPress-iOS
# Add blog picker to Gutenberg editor

Fixes part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/182

This change allows changing target blog while creating a new post:

<img width="388" alt="screen shot 2018-11-22 at 15 13 15" src="https://user-images.githubusercontent.com/5032900/48902358-68154000-ee69-11e8-8c37-046198693d09.png">

**Implementation Details**

Please read [parent PR's](https://github.com/wordpress-mobile/WordPress-iOS/pull/10509) implementation details first.

PostEditorUtil is refactored as a PublishablePostEditor extension so we don't have a PostEditorUtil anymore.

And all PublishablePostEditor extensions are extracted to their own files according to their purpose.

So with this change we start to use PublishablePostEditor extensions to make aztec and gutenberg share same functionality.

The reason of doing this is PostEditorUtil and newly added PublishablePostEditor extensions are sharing the same purpose, they are both holding the logic that aztec and gutenberg have in common. So it is inconsistent to some of it in class and some of it in extensions. That's why I converted PostEditorUtil into an extension too.

**To test:** 

**Prerequisites:**

- clean pods folder
- run "rake dependencies"
- run yarn start on your mobile-gutenberg path
- open XCode and Option+Click Play(a.k.a Run) button
- Select "Run" option on the left
- Select Arguments on right and then check gutenberg option
- Click Run

**1. Test "Save as draft" - Change Target Blog**

- Open "Blog Posts" and tap (+) to create a new post
- Tap Blog picker button in the navigation bar
- Verify blog picker is opened
- Select a different site than current site
- Tap (...) -> Save as Draft
- Verify that Site picker button is turned into a label and you can no longer open the picker from there
- Tap Close
- Switch to that site
- Verify that the draft is added there

**2. Test "Save as draft" - Cancel Picker**

- Open "Blog Posts" and tap (+) to create a new post
- Tap Blog picker button in the navigation bar
- Tap cancel
- Verify that picker is closed

**3. Test "Publish" - Change Target Blog**

- Open "Blog Posts" and tap (+) to create a new post
- Tap Blog picker button in the navigation bar
- Select a different site than current site
- Tap publish
- Verify that post is closed
- Switch to that site
- Verify that the post is published there

**Aztec test**

- Deselect the gutenberg option in program arguments in Xcode and repeat the above 3 tests
